devcycle_python_sdk/managers/config_manager.py

```diff
@@ -1,27 +1,72 @@
+import threading
+import time
 import logging
+import json
 
 from devcycle_python_sdk.dvc_options import DevCycleLocalOptions
 from devcycle_python_sdk.api.local_bucketing import LocalBucketing
+from devcycle_python_sdk.api.config_client import ConfigAPIClient
+from devcycle_python_sdk.exceptions import CloudClientUnauthorizedError, CloudClientError
 
 logger = logging.getLogger(__name__)
 
 
-class EnvironmentConfigManager:
+class EnvironmentConfigManager(threading.Thread):
     def __init__(self, sdk_key: str, options: DevCycleLocalOptions, local_bucketing: LocalBucketing):
-        self.sdk_key = sdk_key
-        self.options = options
-        self.local_bucketing = local_bucketing
+        super().__init__()
 
-        # initialize Config API client
-        # TODO
-        # setup polling thread
-        # add initialized callback
+        self._sdk_key = sdk_key
+        self._options = options
+        self._local_bucketing = local_bucketing
+
+        self._config: dict = None
+        self._config_etag: str = None
+
+        self._config_api_client = ConfigAPIClient(self._sdk_key, self._options)
+
+        self._polling_enabled = True
+        self.daemon = True
+        self.start()
 
     def is_initialized(self) -> bool:
-        return True
+        return self._config is not None
+
+    def get_config(self):
+        try:
+            new_config, new_etag = self._config_api_client.get_config(config_etag=self._config_etag)
+
```

Comment:
Below is related to my comment in config_client.py above.

It is how I'm reacting to the HTTPStatus.NOT_MODIFIED response. I'm not 100% sold on this implementation and open to feedback

Reply:
I think it's easy to follow at least.
Is there an edge case here where we get a Not Modified back but with a matching etag?

Reply:
good question.  so in config_client I am detecting the HTTPStatus.NOT_MODIFIED and always returning the etag that was provided to the getConfig(), so it should always match. 

I'll add an else where I just check that new_config is none and maybe throw a warning and return